server/src/main/java/org/opensearch/cluster/health/ClusterStateHealth.java

```diff
@@ -93,6 +93,17 @@ public ClusterStateHealth(final ClusterState clusterState, final String[] concre
         numberOfDataNodes = clusterState.nodes().getDataNodes().size();
         hasDiscoveredClusterManager = clusterState.nodes().getClusterManagerNodeId() != null;
         indices = new HashMap<>();
+
+        boolean hasSearchOnlyIndices = false;
+        for (String index : concreteIndices) {
```

Comment:
we already iterate over concreteIndices below on line 96, lets set this flag from there?